tests/test_solvers_fairflow.py

```diff
@@ -44,6 +44,58 @@ def test_solvers_fairflow_random():
     # (i.e. overwhelmingly likely to be different)
     assert not np.array_equal(solver_A.affinity_matrix, solver_B.affinity_matrix)
 
+def test_solvers_fairflow_custom_demands():
+    aggregate_score_matrix_A = np.transpose(np.array([
+        [0.2, 0.1, 0.4],
+        [0.5, 0.2, 0.3],
+        [0.2, 0.0, 0.6],
+        [0.7, 0.9, 0.3]
+    ]))
+    constraint_matrix = np.zeros(np.shape(aggregate_score_matrix_A))
+    solver_A = FairFlow(
+        [1,1,1,1],
+        [2,2,2,2],
+        [2,1,3],
```

Comment:
can you add a test where one paper has demand=0 and check there are no assignments for that paper?

Reply:
Done.